pyupgrade/_plugins/lru_cache.py

```diff
@@ -47,13 +47,27 @@ def visit_Call(
             node.func.attr == 'lru_cache' and
             isinstance(node.func.value, ast.Name) and
             node.func.value.id == 'functools' and
-            not node.args and
-            len(node.keywords) == 1 and
-            node.keywords[0].arg == 'maxsize' and
-            isinstance(node.keywords[0].value, ast.NameConstant) and
-            node.keywords[0].value.value is None
+            not node.args
```

Comment:
I think it would be cleaner to rewrite this as `not node.args and _eligible(node.keywords)` and the implementation of `_eligible` would check the kwargs to ensure they fit the criteria for rewriting, which is something like:

1) 1 kwarg and it's `maxsize=None`
2) 2 kwargs and they're `maxsize=None, typed=False` in any order

Reply:
There's also a broader question of whether or not we want to do this rewrite (I think it makes sense but will defer to Anthony)

Reply:
```python

def _eligible(keywords: List[ast.keyword]) -> bool:
    if len(keywords) == 1:
        return _is_literal_kwarg(keywords[0], 'maxsize', None)
    elif len(keywords) == 2:
        return (
            (
                _is_literal_kwarg(keywords[0], 'maxsize', None) and
                _is_literal_kwarg(keywords[1], 'typed', False)
            ) or
            (
                _is_literal_kwarg(keywords[1], 'maxsize', None) and
                _is_literal_kwarg(keywords[0], 'typed', False)
            )
        )
    else:
        return False


def _is_literal_kwarg(
        keyword: ast.keyword, name: str, value: Optional[bool],
) -> bool:
    return (
        keyword.arg == name and
        isinstance(keyword.value, ast.NameConstant) and
        keyword.value.value is value
    )

```

Reply:
Indeed, that requires a couple of extra tests to get the coverage up, but I agree that it's cleaner / more readable. Thanks! I've added you as co-author to the commit